tests/unit/scripts/test_config.py

```diff
@@ -197,3 +203,62 @@ def test_cylc_site_conf_path_env_var(
     GlobalConfig.get_inst()
 
     assert capload == files
+
+
+@pytest.mark.parametrize(
+    'wid_given',
+    [
+        None,
+        'Foo'
+    ]
+)
+@pytest.mark.parametrize(
+    'print_platform_name_set',
+    [
+        True,
+        False
+    ]
+)
+@pytest.mark.parametrize(
+    'print_platform_meta_set',
+    [
+        True,
+        False
+    ]
+)
+def test_options_are_valid(
+    caplog, wid_given, print_platform_name_set, print_platform_meta_set
+):
+    """Test that bad sets of options are not allowed."""
+
+    # Set up truth table for outputs:
+    if print_platform_meta_set and print_platform_name_set:
+        output = False
+        log = '--platform-names is not compatible with --platform-meta'
+    elif (
+        wid_given is not None
+        and (print_platform_name_set or print_platform_meta_set)
+    ):
```

Comment:
I feel like this is replicating the logic of the functions being tested. Better to parametrize in terms of the expected outputs for given inputs, if possible.

Reply:
I think that I've fixed this.